src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -82,31 +85,36 @@ def isEnabled(self, service):
         return enabled
 
     @private
-    def tryConnect(self, host, port):
+    def tryConnect(self, host, port, fnldap):
         max_tries = 3
         connected = False
 
-        timeout = _fs().middlewared.plugins.service_monitor.socket_timeout
+        if self.name == 'activedirectory':
+            host_list = []
 
-        for i in range(0, max_tries):
-            # XXX What about UDP?
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(timeout)
```

Comment:
`def port_is_listening(host, port, errors=[]):` looks very similar to the given code, except it doesn't set any timeout for the socket connection. It may be reasonable to add:
```    def port_is_listening(host, port, timeout=0, errors=[]):
       ....
       if timeout:
          s.settimeout(timeout)
```
in the `freenasldap.py`.

Reply:
Okay. Made changes.